rag/app/book.py

```
@@ -166,13 +166,22 @@ def chunk(filename, binary=None, from_page=0, to_page=MAXIMUM_PAGE_NUMBER, lang=
-        sections = [s.split("@") for s, _ in sections]
-        sections = [(pr[0], "@" + pr[1]) if len(pr) == 2 else (pr[0], "") for pr in sections]
-        chunks = naive_merge(sections, parser_config.get("chunk_token_num", 256), parser_config.get("delimiter", "\n。；！？"))
+        # Keep the "@@" coordinate tag in the section text; it is extracted by
+        # crop() and stripped by remove_tag() downstream in tokenize_chunks.
+        # Pass "" for the legacy layoutno slot so naive_merge's
+        # _reconstruct_text_chunk does not append a stray layout number to the
+        # chunk text. hierarchical_merge drops layoutno the same way (it keeps
+        # only the text), so the two branches stay consistent and the fix
+        # changes nothing in the displayed chunk text except restoring the lost
+        # highlight.
+        chunks = naive_merge([(t, "") for t, _ in sections], parser_config.get("chunk_token_num", 256), parser_config.get("delimiter", "\n。；！？"))
```

Ruff (0.16.1)

[warning] String contains ambiguous `；` (FULLWIDTH SEMICOLON). Did you mean `;` (SEMICOLON)?

(RUF001)